### PR TITLE
Story#599

### PR DESCRIPTION
**Pull request by srikanthgudipudi**

### CODE REVIEW

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts [hunk 1]
```diff
@@ -28,4 +29,8 @@ export class HeadersComponent implements OnInit {
     this.logoutClick.emit('logout');
   }
 
+  menuClick() {
+    this.sideNavClick.emit('');
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts]: dont send it blank, send  "sideMenuToggle"

File: projects/shikshalokam/src/lib/core-module/services/community-service/community.service.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+@Injectable({
+  providedIn: 'root'
+})
+export class CommunityService extends CommonHttpService {
+
+  baseUrl: string;
+
+  constructor(public http: HttpClient) { 
+    super(http);
+    this.baseUrl = 'https://devhome.shikshalokam.org/assessment-service/api/v1';
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/services/community-service/community.service.ts]: It should not be hardcoded

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts (hunk 1 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts]: changed

File: projects/shikshalokam/src/lib/core-module/services/community-service/community.service.ts (hunk 2 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/services/community-service/community.service.ts]: changed

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts [hunk 3]
```diff
@@ -28,4 +29,8 @@ export class HeadersComponent implements OnInit {
     this.logoutClick.emit('logout');
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts:29]: Why are we doing like this??

File: projects/shikshalokam/src/lib/core-module/components/page-layout/page-layout.component.ts [hunk 4]
```diff
@@ -8,11 +11,25 @@ import { MenuData } from '../../interfaces/menu.interface';
...
   type: any;
+  username: any;
   @Input() sideMenuData: MenuData[];
+  @Input() logindetails: any;
   @Output() sideMenuClick = new EventEmitter();
-  constructor() { }
+  @ViewChild('sidenav') public sidenav: MatSidenav;
+
+  mobileQuery: MediaQueryList;
+  private _mobileQueryListener: () => void;
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/components/page-layout/page-layout.component.ts]: remove unwanted codes

File: projects/shikshalokam/src/lib/core-module/services/common-http.service.ts [hunk 5]
```diff
@@ -8,12 +8,17 @@ import { throwError } from 'rxjs';
...
+  http: HttpClient;
+
+  constructor(http: HttpClient) {
+    this.http = http;
   }
 
   // common get method 
   get(apiEndPoint) {
-    return this.http.get(apiEndPoint)
+    return this.http.get(this.baseUrl + apiEndPoint)
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/services/common-http.service.ts]: Where are we setting apiBaseUrl ?

File: projects/shikshalokam/src/lib/core-module/services/config.service.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export const BaseConfig = {
+    baseURL: 'https://devhome.shikshalokam.org/',
```
Comment (kiranharidas187) [inline projects/shikshalokam/src/lib/core-module/services/config.service.ts:2]: It should not be hardcoded. It should come from portal

File: projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts (hunk 3 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/components/headers/headers.component.ts:29]: keycloak is in portal, that is why we are triggering on logout event in npm library and doing logout in portal

File: projects/shikshalokam/src/lib/core-module/components/page-layout/page-layout.component.ts (hunk 4 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/components/page-layout/page-layout.component.ts]: Removed

File: projects/shikshalokam/src/lib/core-module/services/common-http.service.ts (hunk 5 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/services/common-http.service.ts]: Removed

File: projects/shikshalokam/src/lib/core-module/services/config.service.ts (hunk 6 above)
Comment (srikanthgudipudi) [reply projects/shikshalokam/src/lib/core-module/services/config.service.ts:2]: Changed

**Pull request merged**